internal/service/domain/tree/handle_new_sensor_data.go

```diff
@@ -0,0 +1,38 @@
+package tree
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/green-ecolution/green-ecolution-backend/internal/entities"
+	"github.com/green-ecolution/green-ecolution-backend/internal/logger"
+	"github.com/green-ecolution/green-ecolution-backend/internal/service/domain/utils"
+	"github.com/green-ecolution/green-ecolution-backend/internal/storage/postgres/tree"
+)
+
+func (s *TreeService) HandleNewSensorData(ctx context.Context, event *entities.EventNewSensorData) error {
+	log := logger.GetLogger(ctx)
+	log.Debug("handle event", "event", event.Type(), "service", "TreeService")
+	t, err := s.treeRepo.GetBySensorID(ctx, event.New.SensorID)
+	if err != nil {
+		log.Error("failed to get tree by sensor id", "sensor_id", event.New.SensorID, "err", err)
+		return nil
+	}
+
+	status := utils.CalculateWateringStatus(ctx, t.PlantingYear, event.New.Data.Watermarks)
+
+	if status == t.WateringStatus {
+		return nil
+	}
```

Comment:
Maybe add a debug log that nothing has changed?

```suggestion
	if status == t.WateringStatus {
		log.Debug("sensor status has not changed", "sensor_status", status)
		return nil
	}
```